tests/compiler/t_foreach.test.ts

```diff
@@ -105,6 +105,64 @@ describe("t-foreach", () => {
     expect(renderToString(template, context)).toBe(expected);
   });
 
+  test("iterate, Map param", () => {
+    const template = `
+      <t t-foreach="value" t-as="item" t-key="item_index">
+        [<t t-esc="item_index"/>: <t t-esc="item"/> <t t-esc="item_value"/>]
+      </t>`;
+    const expected = ` [0: 1 a]  [1: 2 b]  [2: 3 c] `;
+    const context = {
+      value: new Map([
+        ["a", 1],
+        ["b", 2],
+        ["c", 3],
+      ]),
+    };
```

Comment:
Hello,

I'm a bit confused with this test, I believe this should be "a 1", "b 2" and "c 3", as stated by the doc "for objects and maps, it provides the value (where `$as`  provides the key)"

It comes from
```ts
  } else if (collection instanceof Map) {
    keys = [...collection.keys()];
    values = [...collection.values()];
```

where keys and values are not "inverted", as opposed to what is done for the object case?

Reply:
Definitely a mistake, didn't realize the existing code for objects was inverted for some reason... I'll make a fix tomorrow.

Reply:
https://github.com/odoo/owl/pull/1502